cpp/tests/strings/json_tests.cpp

```diff
@@ -0,0 +1,506 @@
+  "\"Nigel Rees\",\n\t\t\t\"title\": \"Sayings of the Century\",\n\t\t\t\"price\": "
+  "8.95\n\t\t},\n\t\t{\n\t\t\t\"category\": \"fiction\",\n\t\t\t\"author\": \"Evelyn "
+  "Waugh\",\n\t\t\t\"title\": \"Sword of Honour\",\n\t\t\t\"price\": "
+  "12.99\n\t\t},\n\t\t{\n\t\t\t\"category\": \"fiction\",\n\t\t\t\"author\": \"Herman "
+  "Melville\",\n\t\t\t\"title\": \"Moby Dick\",\n\t\t\t\"isbn\": "
+  "\"0-553-21311-3\",\n\t\t\t\"price\": 8.99\n\t\t},\n\t\t{\n\t\t\t\"category\": "
+  "\"fiction\",\n\t\t\t\"author\": \"J. R. R. Tolkien\",\n\t\t\t\"title\": \"The Lord of the "
+  "Rings\",\n\t\t\t\"isbn\": \"0-395-19395-8\",\n\t\t\t\"price\": "
+  "22.99\n\t\t}\n\t],\n\t\"bicycle\": {\n\t\t\"color\": \"red\",\n\t\t\"price\": "
+  "19.95\n\t}\n},\n\"expensive\": 10\n}"};
+
+std::unique_ptr<cudf::column> drop_whitespace(cudf::column_view const& col)
+{
+  cudf::test::strings_column_wrapper whitespace{"\n", "\r", "\t"};
+  cudf::test::strings_column_wrapper repl{"", "", ""};
+
+  cudf::strings_column_view strings(col);
+  cudf::strings_column_view targets(whitespace);
+  cudf::strings_column_view replacements(repl);
+  return cudf::strings::replace(strings, targets, replacements);
+}
+
+struct JsonTests : public cudf::test::BaseFixture {
+};
+
+TEST_F(JsonTests, GetJsonObjectRootOp)
+{
+  // root
+  cudf::test::strings_column_wrapper input{json_string};
+  std::string json_path("$");
+  auto result_raw = cudf::strings::get_json_object(cudf::strings_column_view(input), json_path);
+  auto result     = drop_whitespace(*result_raw);
+
+  auto expected = drop_whitespace(input);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*result, *expected);
+}
+
+TEST_F(JsonTests, GetJsonObjectChildOp)
+{
+  {
+    cudf::test::strings_column_wrapper input{json_string};
+    std::string json_path("$.store");
+    auto result_raw = cudf::strings::get_json_object(cudf::strings_column_view(input), json_path);
+    auto result     = drop_whitespace(*result_raw);
+
+    cudf::test::strings_column_wrapper expected_raw{
+      "{\n\t\"book\": [\n\t\t{\n\t\t\t\"category\": \"reference\",\n\t\t\t\"author\": "
+      "\"Nigel Rees\",\n\t\t\t\"title\": \"Sayings of the Century\",\n\t\t\t\"price\": "
+      "8.95\n\t\t},\n\t\t{\n\t\t\t\"category\": \"fiction\",\n\t\t\t\"author\": \"Evelyn "
+      "Waugh\",\n\t\t\t\"title\": \"Sword of Honour\",\n\t\t\t\"price\": "
+      "12.99\n\t\t},\n\t\t{\n\t\t\t\"category\": \"fiction\",\n\t\t\t\"author\": \"Herman "
+      "Melville\",\n\t\t\t\"title\": \"Moby Dick\",\n\t\t\t\"isbn\": "
+      "\"0-553-21311-3\",\n\t\t\t\"price\": 8.99\n\t\t},\n\t\t{\n\t\t\t\"category\": "
+      "\"fiction\",\n\t\t\t\"author\": \"J. R. R. Tolkien\",\n\t\t\t\"title\": \"The Lord of the "
+      "Rings\",\n\t\t\t\"isbn\": \"0-395-19395-8\",\n\t\t\t\"price\": "
+      "22.99\n\t\t}\n\t],\n\t\"bicycle\": {\n\t\t\"color\": \"red\",\n\t\t\"price\": "
+      "19.95\n\t}\n}"};
```

Comment:
Maybe reformat these too? This is very difficult to visually parse and therefore hard to review.